### PR TITLE
Fix: improve watching for changes in ancillary resources

### PR DESCRIPTION
**Pull request by amisevsk**
### What does this PR do?
1. Mark the controller as `Own`ing configmaps, secrets, and serviceaccounts, in order to ensure reconciles are triggered when these objects are modified
2. Restrict DWO configmap syncs to only events involving that configmap (instead of all configmaps)

### What issues does this PR fix or reference?
1. Editing e.g. the `workspaceID-metadata` configmap doesn't trigger a reconcile (state is restored on next devworkspace reconcile)
2. Probably harmless but frequent syncs with the `devworkspace-controller-configmap` (check by putting a `Printf` in `updateConfigMap`) -- previously, any event relating to any configmap would trigger a reconcile 

### Is it tested? How?
Start a devworkspace and try to edit the `metadata` configmap (or an owned secret/the sa)

### PR Checklist

- [ ] E2E tests pass (when PR is ready, comment `/test v8-devworkspace-operator-e2e, v8-che-happy-path` to trigger)
    - [ ] `v8-devworkspace-operator-e2e`: DevWorkspace e2e test
    - [ ] `v8-che-happy-path`: Happy path for verification integration with Che

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -242,11 +242,15 @@ func fillOpenShiftRouteSuffixIfNecessary(nonCachedClient client.Client, configMa
 func ConfigMapPredicates(mgr manager.Manager) predicate.Predicate {
 	return predicate.Funcs{
 		UpdateFunc: func(evt event.UpdateEvent) bool {
-			updateConfigMap(mgr.GetClient(), evt.ObjectNew)
+			if evt.ObjectNew.GetName() == "devworkspace-controller-configmap" {
```
Comment (sleshchenko) [inline pkg/config/config.go:245]: please, let's use it from constant

**Review by sleshchenko — APPROVED** [review]
Looks meaningful

File: pkg/config/config.go [hunk 2]
```diff
@@ -242,11 +242,15 @@ func fillOpenShiftRouteSuffixIfNecessary(nonCachedClient client.Client, configMa
...
+				updateConfigMap(mgr.GetClient(), evt.ObjectNew)
```
Comment (sleshchenko) [inline pkg/config/config.go:246]: well, let's rename updateConfigMap because it's confusing. Like `loadConfigFromCM`

**Review by JPinkney — APPROVED** [review]

File: pkg/config/config.go (hunk 2 above)
Comment (amisevsk) [reply pkg/config/config.go]: Good point -- we're not really watching anything there anymore.

File: pkg/config/config.go (hunk 1 above)
Comment (amisevsk) [reply pkg/config/config.go]: Yeah, accidentally left the "lets see if this works" implementation here. Updated to match name/namespace from `ConfigMapReference` (which can be overridden via env var)

**Review by JPinkney — APPROVED** [review]

**Review by sleshchenko — APPROVED** [review]

**Pull request merged**